mmpy_bot/plugins/base.py

```diff
@@ -25,39 +27,19 @@ class Plugin(ABC):
 
     def __init__(self):
         self.driver: Optional[Driver] = None
-        self.message_listeners: Dict[
-            re.Pattern, Sequence[MessageFunction]
-        ] = defaultdict(list)
-        self.webhook_listeners: Dict[
-            re.Pattern, Sequence[WebHookFunction]
-        ] = defaultdict(list)
-
-        # We have to register the help function listeners at runtime to prevent the
-        # Function object from being shared across different Plugins.
-        self.help = listen_to("^help$", needs_mention=True)(Plugin.help)
-        self.help = listen_to("^!help$")(self.help)
-
-    def initialize(self, driver: Driver, settings: Optional[Settings] = None):
-        self.driver = driver
-
-        # Register listeners for any listener functions we might have
-        for attribute in dir(self):
-            attribute = getattr(self, attribute)
-            if isinstance(attribute, Function):
-                # Register this function and any potential siblings
-                for function in [attribute] + attribute.siblings:
-                    function.plugin = self
-                    if isinstance(function, MessageFunction):
-                        self.message_listeners[function.matcher].append(function)
-                    elif isinstance(function, WebHookFunction):
-                        self.webhook_listeners[function.matcher].append(function)
-                    else:
-                        raise TypeError(
-                            f"{self.__class__.__name__} has a function of unsupported"
-                            f" type {type(function)}."
-                        )
+        self.plugin_manager: Optional[PluginManager] = None
+        self.settings: Optional[Settings] = None
+        self.docstring = self.__doc__ if self.__doc__ != Plugin.__doc__ else None
 
-        return self
+    def initialize(
+        self,
+        driver: Driver,
+        plugin_manager: PluginManager,
+        settings: Settings,
+    ):
+        self.driver = driver
+        self.plugin_manager = plugin_manager
+        self.settings = settings
```

Comment:
```suggestion
        self.settings = settings
```

The `Plugin` does not actually need the manager for anything, only the `HelpPlugin` does, and therefore that one can just override this function (which it already does anyway)